controlplane/internal/upgrade/upgrade.go

```diff
@@ -151,6 +155,34 @@ func (u *OpenshiftUpgrader) UpdateClusterVersionDesiredUpdate(ctx context.Contex
 	return nil
 }
 
+func (u *OpenshiftUpgrader) VerifyUpgradedNodes(ctx context.Context, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) error {
```

Comment:
Shouldnt this be handled by https://github.com/openshift-assisted/cluster-api-agent/blob/92cebf02e87613e0194b807776e3abde5ecca98e/controlplane/internal/controller/openshiftassistedcontrolplane_controller.go#L536C51-L536C70?

Reply:
Ideally it would be, but we can't actually update the machines so this a work-around

Reply:
I mean that even if we want to run the workaround, I think we should keep the logic updating those fields together (we should make sure it runs that part of reconciliation during upgrades, but still I don't think we should split the logic handling those fields)